web_sdk/web_test_utils/lib/image_compare.dart

```diff
@@ -9,24 +9,45 @@ import 'package:path/path.dart' as p;
 
 import 'environment.dart';
 import 'goldens.dart';
+import 'skia_client.dart';
 
-/// Compares a screenshot taken through a test with it's golden.
+/// Whether this code is running on LUCI.
+bool _isLuci = Platform.environment.containsKey('SWARMING_TASK_ID') && Platform.environment.containsKey('GOLDCTL');
+bool _isPreSubmit = _isLuci && Platform.environment.containsKey('GOLD_TRYJOB');
+bool _isPostSubmit = _isLuci && !_isPreSubmit;
+
+
+/// Compares a screenshot taken through a test with its golden.
 ///
 /// Used by Flutter Web Engine unit tests and the integration tests.
 ///
 /// Returns the results of the tests as `String`. When tests passes the result
 /// is simply `OK`, however when they fail it contains a detailed explanation
 /// on which files are compared, their absolute locations and an HTML page
 /// that the developer can see the comparison.
-String compareImage(
+Future<String> compareImage(
   Image screenshot,
   bool doUpdateScreenshotGoldens,
   String filename,
   PixelComparison pixelComparison,
-  double maxDiffRateFailure, {
+  double maxDiffRateFailure,
+  SkiaGoldClient? skiaClient, {
+  // TODO(mdebbar): Remove these args with goldens repo.
   String goldensDirectory = '',
+  String filenameSuffix = '',
   bool write = false,
-}) {
+}) async {
+  if (_isLuci && skiaClient != null) {
+    // This is temporary to get started by uploading existing screenshots to
+    // Skia Gold. The next step would be to actually use Skia Gold for
+    // comparison.
+
+    // TODO(mdebbar): Use Skia Gold for comparison, not only for uploading.
```

Comment:
Might be good to add to this comment that when this is being used for comparison, it should account for any auto rollers, they will break if it fails in presubmit. We should be able to really easily add support of the flutter-gold check to manage checking in images when its ready, just a small adjustment in cocoon. :)

Reply:
#### 1. Autorollers:
We plan to rely on the gold check for failures in pre-submit (we won't fail any tests even if the golden match fails). This comment/TODO was meant for local testing. Sorry that wasn't clear.

#### 2. gold-check & cocoon:
Out of the box, we got those comments by the gold bot that provide triage links. I looked at the cocoon repo and found a couple places that need to be adjusted to work for the engine. Is there anything else we need to do (besides the cocoon change) to get the extra gold check to appear on engine PRs?